### ISSUE TITLE
Add option to enter Google Maps Geocode API 

### ISSUE THREAD

**Issue by arnaudbroes** [semperfiwebdesign/simplemap#276]
For customers who add large amounts of locations at a time. The free quota is 2,500 queries per day.
https://developers.google.com/maps/documentation/geocoding/start

**Comment by wpsmort** [comment]
Regarding this post - http://support.simplemap-plugin.com/discussions/questions/1535-your-servers-ip-is-over-the-geocode-threshold-set-by-google#comment_46527286

It looks like we need to add a second Google Maps API Key field.  One API key is required for the client side JavaScript API and one key is required for the server side web service.

**Comment by wpsmort** [comment]
Google Maps is returning this error - 

Geocoding Service: This API project is not authorized to use this API.  For more information on authentication and Google Maps JavaScript API services please see: https://developers.google.com/maps/documentation/javascript/get-api-key

This may be related to a known issue with recent changes by Google to their Maps API.  It looks like they may now need a separate API key in order to access their Geocoding Service API.  We're currently looking into this and, if another API key is required, we'll release an update to the plugin so users can add this additional API key.

**Comment by arnaudbroes** [comment]
@wpsmort can you provide me with an API key with those specific restrictions via Slack again? I'm afraid I can't set up Billing in the Google Cloud Platform because Google doesn't accept my credit card.

**Comment by wpsmort** [comment]
I've provided training to Arnaud on how to set up a Google Maps API Key and he's creating documentation to go on the website.  

I've tested SimpleMap with a properly created API Key and a location does get geocoded when I add one.  However, I do get a message that says "Your Server's IP is over the geocode threshold set by Google so we had to fallback to a Javascript function."  See screenshot below.

<img width="1038" alt="screen shot 2019-01-10 at 5 50 45 pm" src="https://user-images.githubusercontent.com/17548525/51004325-1d7a8080-1508-11e9-93fe-6dfe6a90dee6.png">

The message is coming from here - https://github.com/semperfiwebdesign/simplemap/blob/a6d5fc36cc21db9dec126dc7b61183053683aa3a/classes/locations.php#L244.  We should determine if this message is valid (I don't believe it is valid) and what is triggering it.  We should then bee able to determine if we need to an option for users to enter a Geocode API Key or not (I don't believe that we do).

**Comment by arnaudbroes** [comment]
@wpsmort correct, that message is only partially valid. It's echoed when the PHP geocoding fails. It's too specific and should be a generic message, because the PHP geocoding is failing for a different reason.
